src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1270,9 +1272,9 @@ internal static void ValidateFunction(FunctionDescriptor function, Dictionary<st
             }
         }
 
-        internal static void ValidateHttpFunction(string functionName, HttpTriggerAttribute httpTrigger)
+        internal static void ValidateHttpFunction(string functionName, HttpTriggerAttribute httpTrigger, bool isProxy = false)
         {
-            if (string.IsNullOrWhiteSpace(httpTrigger.Route))
+            if (string.IsNullOrWhiteSpace(httpTrigger.Route) && !isProxy)
```

Comment:
Why do we want to skip validation for proxies here?

Reply:
In case of the route to the root, the Route property is empty and we don't want the function name to be assigned as the value of the Route if it is a proxy function descriptor. We will still need the value to be empty so the route engine can correctly map the incoming request.

Reply:
Just worried about not checking the routes for the `/admin` endpoint. We want to fail if any proxies try to use that.

Reply:
/admin endpoint uses its own Conttoller and never runs this code path which will be called from FunctionController.
 Plus the check for proxy is only when the route is empty.

So /admin/* endpoints never run the proxy code. But that's a good point and I will add a test to proxy integration tests to make sure /admin/* never gets into proxy logic.

Reply:
My point is mostly about appropriately tracing and disallowing proxy routes with that, even though actual requests will be picked up by the other controllers with routes under `admin`, this validation is performed on initialization and it's a chance to provide feedback on the configuration issue.

Reply:
Not sure if I understand what you mean by provide feedback on the configuration issue?
If the request is to /admin/* then the Route property even if it is returned from Proxy will not be empty. It will only be empty if the request is to the root.